pkg/metricscollector/opentelemetry.go

```diff
@@ -78,37 +95,107 @@ func initCounter() {
 	if err != nil {
 		otLog.Error(err, msg)
 	}
-}
 
-func (o *OtelMetrics) RecordScalerMetric(namespace string, scaledObject string, scaler string, scalerIndex int, metric string, value float64) {
-	cback := func(ctx context.Context, obsrv api.Float64Observer) error {
-		obsrv.Observe(value, getScalerMeasurementOption(namespace, scaledObject, scaler, scalerIndex, metric))
-		return nil
-	}
-	_, err := meter.Float64ObservableGauge(
+	_, err = meter.Float64ObservableGauge(
 		"keda.scaler.metrics.value",
 		api.WithDescription("Metric Value used for HPA"),
-		api.WithFloat64Callback(cback),
+		api.WithFloat64Callback(ScalerMetricValueCallback),
 	)
```

Comment:
We are still creating a lot of observable gauges. As it's observable, I'd say that we just need to init it once (or maybe once per label combination) and just update the referenced variable where the value is stored. 
I'm not 100% who the observable stuff work in go's OTEL SDK, but in dotnet SDK when you register the observable metric, it executes the callback internally on each iteration, so you should not register the same observable item more than once

Reply:
Yes. That's why I move them to `initMeters()` function. Is there any gauge I missed?

Reply:
you're right, sorry :(
I've misunderstood the diff generated by GH 😥